### PR TITLE
New app: Matrix Synapse

### PR DESCRIPTION
**Pull request by Schiriki123**
App Suggestion: #283 

- [x] I have tested the template using the method described in README.md thoroughly
- [x] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [x] Icon is added as a png file to the logos directory.
- [x] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [ ] I have made sure that instructions.start and instructions.end are clear and self-explanatory.

For the instructions I would like to get some feedback.
If something is unclear or doesn't work please comment here.

### CODE REVIEW

**Comment by Schiriki123** [comment]
There should be a  better solution for matrix synapse. didn't saw that you can use "dockerfileLines"

**Comment by Schiriki123** [comment]
Please don't merge yet, I will improve the script when I have the time

**Comment by Schiriki123** [comment]
This should work now

File: public/v4/apps/matrix-synapse.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      validRegex: '/^[0-9]{1,}|latest$/'
+    - id: '$$cap_db_password'
+      label: Database password
+      defaultValue: $$cap_gen_random_hex(20)
+      description: >-
+          The password for the Synapse database.
+      validRegex: '/^[0-9a-zA-Z]{1,}$/'
+  instructions:
+    start: >-
+      Matrix synapse server
```
Comment (githubsaturn) [inline public/v4/apps/matrix-synapse.yml:86]: Please add more context here. A link to the website, a quick introduction etc... See [plex](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/plex.yml#L44-L47) for example

**Review by githubsaturn — COMMENTED** [review]
Just to double check, this is fully tested at this point, right?

PS: please run the formatter to make the build pass. You can do so by running `npm run formatter-write`

**Comment by asitav-sen** [comment]
Hi,
Can you please help me with this? running
`sudo docker exec -it $(sudo docker ps | grep -o srv-captain--matrix.*) register_new_matrix_user http://localhost:8008 -c /data/homeserver.yaml`
throws error
`OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "srv-captain--matrix-db.1.2bc7v9cz9xjiwroc5xih53zz2": executable file not found in $PATH: unknown`
Thanks.

**Comment by Schiriki123** [comment]
You can try to enter the container name manually like this: 
`sudo docker exec -it CONTAINER_NAME register_new_matrix_user http://localhost:8008 -c /data/homeserver.yaml`

You can get the name of the container by running `sudo docker container ls` and look for a container with the given name, but without the .db at the end

Some thing like this:
`sudo docker exec -it srv-captain--matrix.1.2bc7v9cz9xjiwroc5xih53zz2 register_new_matrix_user http://localhost:8008 -c /data/homeserver.yaml`

**Comment by asitav-sen** [comment]
> You can try to enter the container name manually like this: `sudo docker exec -it CONTAINER_NAME register_new_matrix_user http://localhost:8008 -c /data/homeserver.yaml`
> 
> You can get the name of the container by running `sudo docker container ls` and look for a container with the given name, but without the .db at the end
> 
> Some thing like this: `sudo docker exec -it srv-captain--matrix.1.2bc7v9cz9xjiwroc5xih53zz2 register_new_matrix_user http://localhost:8008 -c /data/homeserver.yaml`

Thanks @Schiriki123 . It worked.

**Comment by Schiriki123** [comment]
I created an PR #806 that should fix this issue

**Pull request merged**